src/main/java/mil/nga/tiff/Rasters.java

```diff
@@ -147,39 +132,19 @@ private void validateValues() {
 	 *            width of pixels
 	 * @param height
 	 *            height of pixels
-	 * @param samplesPerPixel
-	 *            samples per pixel
-	 * @param bitsPerSample
-	 *            bits per sample
+	 * @param sampleTypes
+	 *            Sample types per sample
 	 */
-	public Rasters(int width, int height, int samplesPerPixel,
-			List<Integer> bitsPerSample, FieldType[] sampleFieldTypes, ByteOrder order) {
-		this(width, height, samplesPerPixel, bitsPerSample, sampleFieldTypes,
-				new ByteBuffer[samplesPerPixel]);
+	public Rasters(int width, int height, SampleType[] sampleTypes, ByteOrder order) {
+		this(width, height, sampleTypes, new ByteBuffer[sampleTypes.length]);
 		for (int i = 0; i < sampleValues.length; ++i) {
-			sampleValues[i] = ByteBuffer.allocate(width * height * sizeSample(i));
-			sampleValues[i].order(order);
+			sampleValues[i] = ByteBuffer
+								.allocateDirect(width * height * sampleTypes[i].byteSize)
+								.order(order);
 		}
 	}
 
 	/**
-	 * Constructor
-	 * 
-	 * @param width
-	 *            width of pixels
-	 * @param height
-	 *            height of pixels
-	 * @param samplesPerPixel
-	 *            samples per pixel
-	 * @param bitsPerSample
-	 *            bits per sample for all samples of a pixel
-	 */
-	public Rasters(int width, int height, int samplesPerPixel, int bitsPerSample, FieldType[] sampleFieldTypes, ByteOrder order) {
```

Comment:
please add a convenience constructor public Rasters(int width, int height, int samplesPerPixel, SampleType sampleType) that internally uses ByteOrder.nativeOrder() for the byte order and rename makeBitsPerSampleList to makeSampleTypeList(int samplesPerPixel, SampleType sampleType)